letshelp-letsencrypt/letshelp_letsencrypt/letshelp_letsencrypt_apache.py

```diff
@@ -0,0 +1,303 @@
+    print "\nFiles:"
+    for copied_file in copied_files:
+        print copied_file
+    print "Directories (including all contained files):"
+    for copied_dir in copied_dirs:
+        print copied_dir
+
+    sys.stdout.write("\nIs it safe to submit these files? ")
+    while True:
+        ans = raw_input("(Y)es/(N)o: ").lower()
+        if ans.startswith("y"):
+            return
+        elif ans.startswith("n"):
+            sys.exit("Your files were not submitted")
+
+
+def copy_config(server_root, temp_dir):
+    """Safely copies server_root to temp_dir and returns copied files
+
+    :param str server_root: Absolute path to the Apache server root
+    :param str temp_dir: Path to the temporary directory to copy files to
+
+    :returns: List of copied files and a list of leaf directories where
+        all contained files were copied
+    :rtype: `tuple` of `list` of `str`
+
+    """
+    copied_files, copied_dirs = list(), list()
+    dir_len = len(os.path.dirname(server_root))
+
+    for config_path, config_dirs, config_files in os.walk(server_root):
+        temp_path = os.path.join(temp_dir, config_path[dir_len+1:])
+        os.mkdir(temp_path)
+
+        copied_all = True
+        copied_files_in_current_dir = list()
+        for config_file in config_files:
+            config_file_path = os.path.join(config_path, config_file)
+            temp_file_path = os.path.join(temp_path, config_file)
+            if os.path.islink(config_file_path):
+                os.symlink(os.readlink(config_file_path), temp_file_path)
+            elif safe_config_file(config_file_path):
+                copy_file_without_comments(config_file_path, temp_file_path)
+                copied_files_in_current_dir.append(config_file_path)
+            else:
+                copied_all = False
+
+        # If copied all files in leaf directory
+        if copied_all and not config_dirs:
+            copied_dirs.append(config_path)
+        else:
+            copied_files += copied_files_in_current_dir
+
+    return copied_files, copied_dirs
+
+
+def copy_file_without_comments(source, destination):
+    """Copies source to destination, removing comments
+
+    :param str source: Path to the file to be copied
+    :param str destination: Path where source should be copied to
+
+    """
+    with open(source, "r") as infile:
+        with open(destination, "w") as outfile:
+            for line in infile:
+                if not (line.isspace() or line.lstrip().startswith("#")):
+                    outfile.write(line)
+
+
+def safe_config_file(config_file):
+    """Returns True if config_file can be safely copied
+
+    :param str config_file: Path to an Apache configuration file
+
+    :returns: True if config_file can be safely copied
+    :rtype: bool
+
+    """
+    config_file_lower = config_file.lower()
+    if _SENSITIVE_FILENAME_REGEX.search(config_file_lower):
+        return False
+
+    proc = subprocess.Popen(["file", config_file],
```

Comment:
It depends on how this gets used. I wrote the script with the impression that we we  would/could link people to this single file which they could easily download and run to bundle up and send us their Apache configuration. Because of this, I purposely avoided including any modules that aren't in the standard library.

If I was wrong in this assumption and we are going to have people install this script in virtualenv, then I agree `filemagic` makes things MUCH nicer.